### PR TITLE
Bug 1823676: Aggregate conditions from all controllers

### PR DESCRIPTION
**Pull request by dmage**
The ClusterOperator objects represent the entire operator. The image registry operator has several controllers, and if one of them has a problem, the operator should report it.

### CODE REVIEW

File: pkg/client/fake/fixtures.go [hunk 1]
```diff
@@ -70,18 +70,6 @@ func NewFixturesBuilder() *FixturesBuilder {
 	return factory
 }
 
-// AddDaemonSets adds appsv1.DaemonSets to the lister cache
-func (f *FixturesBuilder) AddDaemonSets(objs ...*appsv1.DaemonSet) *FixturesBuilder {
-	for _, v := range objs {
-		err := f.dsIndexer.Add(v)
```
[the remark this answers is not in the document]
Comment (dmage) [reply pkg/client/fake/fixtures.go:76]: removed

File: pkg/client/fake/fixtures.go [hunk 2]
```diff
@@ -186,26 +174,6 @@ func (f *FixturesBuilder) AddClusterRoleBindings(objs ...*rbacv1.ClusterRoleBind
 	return f
 }
 
-// AddImageConfig adds cluster-wide config.openshift.io/v1 Image to the lister cache
-func (f *FixturesBuilder) AddImageConfig(config *configv1.Image) *FixturesBuilder {
-	err := f.imageConfigsIndexer.Add(config)
```
[the remark this answers is not in the document]
Comment (dmage) [reply pkg/client/fake/fixtures.go:191]: removed

File: pkg/client/fake/fixtures.go [hunk 3]
```diff
@@ -186,26 +174,6 @@ func (f *FixturesBuilder) AddClusterRoleBindings(objs ...*rbacv1.ClusterRoleBind
...
-	if err != nil {
-		panic(err)
-	}
-	return f
-}
-
-// AddClusterOperators adds config.openshift.io/v1 ClusterOperators to the lister cache
-func (f *FixturesBuilder) AddClusterOperators(objs ...*configv1.ClusterOperator) *FixturesBuilder {
-	for _, v := range objs {
-		err := f.clusterOperatorsIndexer.Add(v)
```
[the remark this answers is not in the document]
Comment (dmage) [reply pkg/client/fake/fixtures.go:201]: removed

**Comment by ricardomaraschini** [comment]
/retest

Error during cluster deployment.

**Pull request merged**